gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -243,14 +243,28 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
         val kspTaskName = kotlinCompileProvider.name.replaceFirst("compile", "ksp")
 
         val kotlinCompileTask = kotlinCompileProvider.get()
+        val sourceSetOptions = kspConfigurations.resolvedSourceSetOptions(kotlinCompilation)
+
+        fun Task.addKspMetadataDependencyIfExists(sourceSetName: String) {
+            val parentKspTaskName = lowerCamelCased("ksp", sourceSetName, "KotlinMetadata")
+
+            project.locateTask<KspTask>(parentKspTaskName)?.let { parentKspTask ->
+                dependsOn(parentKspTask)
```

Comment:
Leaving out dependencies on parent KSP tasks will make the integration test `KMPWithHmppIT#testCustomSourceSetHierarchyBuild` fail:
```
> Task :workload:compileKotlinJvm FAILED
Execution optimizations have been disabled for task ':workload:compileKotlinJvm' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/tmp/junit389805203877564473/workload/build/generated/ksp/metadata/clientMain/kotlin'. Reason: Task ':workload:compileKotlinJvm' uses this output of task ':workload:kspClientMainKotlinMetadata' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/tmp/junit389805203877564473/workload/build/generated/ksp/metadata/commonMain/kotlin'. Reason: Task ':workload:compileKotlinJvm' uses this output of task ':workload:kspCommonMainKotlinMetadata' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
e: /tmp/junit389805203877564473/workload/src/clientMain/kotlin/com/example/ClientMainAnnotated.kt: (5, 20): Unresolved reference: ClientMainAnnotatedForClientMain
```

The above dependencies seem analogous to this `kotlinCompile.dependsOn(kspTaskProvider)`, which has been present before.

What's your suggestion to do instead?